cmd/soroban-cli/src/bin/main.rs

```diff
@@ -1,11 +1,23 @@
-use clap::CommandFactory;
-use soroban_cli::Root;
+use clap::{CommandFactory, Parser};
+use soroban_cli::{commands::plugin, Root};
 
 #[tokio::main]
 async fn main() {
-    let root = Root::new().unwrap_or_else(|e| {
-        let mut cmd = Root::command();
-        e.format(&mut cmd).exit();
+    let root = Root::try_parse().unwrap_or_else(|e| {
+        if std::env::args().any(|s| &s == "--list") {
```

Comment:
Suggestion: This seems like it ends up handling `--list` anywhere in the command:
e.g. `soroban contract --list`

Which is a bit surprising. Shouldn't this only be hit if there is no subcommand?

Reply:
Like:
```diff
diff --git a/cmd/soroban-cli/src/bin/main.rs b/cmd/soroban-cli/src/bin/main.rs
index ea07f92..be9d2e1 100644
--- a/cmd/soroban-cli/src/bin/main.rs
+++ b/cmd/soroban-cli/src/bin/main.rs
@@ -4,10 +4,6 @@ use soroban_cli::{commands::plugin, Root};
 #[tokio::main]
 async fn main() {
     let root = Root::try_parse().unwrap_or_else(|e| {
-        if std::env::args().any(|s| &s == "--list") {
-            println!("{}", plugin::list().unwrap_or_default().join("\n"));
-            std::process::exit(0);
-        }
         if let clap::error::ErrorKind::InvalidSubcommand = e.kind() {
             if let Err(error) = plugin::run() {
                 eprintln!("error: {error}");
diff --git a/cmd/soroban-cli/src/commands/mod.rs b/cmd/soroban-cli/src/commands/mod.rs
index 829e82b..ee42c35 100644
--- a/cmd/soroban-cli/src/commands/mod.rs
+++ b/cmd/soroban-cli/src/commands/mod.rs
@@ -54,7 +54,7 @@ pub struct Root {
     #[arg(long)]
     pub list: bool,
     #[command(subcommand)]
-    pub cmd: Cmd,
+    pub cmd: Option<Cmd>,
 }
 
 impl Root {
@@ -72,12 +72,20 @@ impl Root {
     }
     pub async fn run(&self) -> Result<(), Error> {
         match &self.cmd {
-            Cmd::Contract(contract) => contract.run().await?,
-            Cmd::Events(events) => events.run().await?,
-            Cmd::Lab(lab) => lab.run().await?,
-            Cmd::Version(version) => version.run(),
-            Cmd::Completion(completion) => completion.run(),
-            Cmd::Config(config) => config.run()?,
+            Some(Cmd::Contract(contract)) => contract.run().await?,
+            Some(Cmd::Events(events)) => events.run().await?,
+            Some(Cmd::Lab(lab)) => lab.run().await?,
+            Some(Cmd::Version(version)) => version.run(),
+            Some(Cmd::Completion(completion)) => completion.run(),
+            Some(Cmd::Config(config)) => config.run()?,
+            None => {
+                if self.list {
+                    println!("{}", plugin::list().unwrap_or_default().join("\n"));
+                } else {
+                    let _ = Self::command().print_help();
+                    std::process::exit(1);
+                }
+            }
         };
         Ok(())
     }
```

Reply:
Fixed it by checking if there is a missing subcommand.  Pushed, but still need to figure out the windows hex string issue.